assets/js/hooks/useQueryArg.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * `useQueryArg` hook.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useCallback } from '@wordpress/element';
+import { addQueryArgs, getQueryArg } from '@wordpress/url';
+
+/**
+ * Uses a location query param as a state.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} key            The query param key to be used.
+ * @param {string} [initialValue] Optional. The initial value for the query param to be used.
+ * @return {Array} The getter and setter for the query param state.
+ */
+function useQueryArg( key, initialValue ) {
```

Comment:
I'd be inclined to add a third argument for dependency injection here, rather than the global mocking done in the test below. Something like `function useQueryArg( key, initialValue, _global = global ) { ` or something. What do you think?

Reply:
This actually is a great idea. Thanks :)